### PR TITLE
[PATCH v6] test: dma_perf: drain poll transfers before teardown

### PR DESCRIPTION
**Pull request by PavanNikhilesh**
Wait for transfers to complete in asynchronous poll mode before
tearing down the associated memory.

Signed-off-by: Pavan Nikhilesh <pbhagavatula@marvell.com>
Reviewed-by: Tuomas Taipale <tuomas.taipale@nokia.com>

### CODE REVIEW

File: test/performance/odp_dma_perf.c [hunk 1]
```diff
@@ -1123,6 +1124,21 @@ static void drain_compl_events(void)
 	}
 }
 
+static void
+drain_poll_transfers(sd_t *sd)
+{
+	const uint32_t count = sd->dma.num_inflight;
+	trs_info_t *infos = sd->dma.infos, *info;
+	odp_dma_t handle = sd->dma.handle;
+	odp_dma_result_t res;
```
Comment (TuomasTaipale) [inline test/performance/odp_dma_perf.c:1133]: Result usage is optional so perhaps `NULL` could be used directly in `odp_dma_transfer_done()` below, making this unnecessary.

File: test/performance/odp_dma_perf.c [hunk 2]
```diff
@@ -1123,6 +1124,21 @@ static void drain_compl_events(void)
...
+
+	for (uint32_t i = 0U; i < count; ++i) {
+		info = &infos[i];
```
Comment (TuomasTaipale) [inline test/performance/odp_dma_perf.c:1136]: There could be a newline after this assignment and before the `if` below.

File: test/performance/odp_dma_perf.c [hunk 3]
```diff
@@ -1123,6 +1124,21 @@ static void drain_compl_events(void)
...
+		if (info->is_running)
+			odp_dma_transfer_done(handle, info->compl_param.transfer_id, &res);
```
Comment (TuomasTaipale) [inline test/performance/odp_dma_perf.c:1138]: Would the draining be more reliable if there was a `while` here? So, looping as long as `odp_dma_transfer_done()` returns 0 ("transfer has not finished").

File: test/performance/odp_dma_perf.c (hunk 2 above)
Comment (PavanNikhilesh) [reply test/performance/odp_dma_perf.c:1136]: Ack, I will modify in v2.

File: test/performance/odp_dma_perf.c [hunk 4]
```diff
@@ -1111,7 +1111,8 @@ static void wait_compl_event(sd_t *sd, stats_t *stats)
 	}
 }
 
-static void drain_compl_events(void)
+static void
+drain_compl_events(ODP_UNUSED sd_t *sd)
```
Comment (TuomasTaipale) [inline test/performance/odp_dma_perf.c:1115]: Existing style in file is to keep return type and name at the same line.

**Review by TuomasTaipale — APPROVED** [review]
One cosmetic comment, otherwise OK, thanks for the patch.

Reviewed-by: Tuomas Taipale <tuomas.taipale@nokia.com>

**Pull request merged**